src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ConnectionsPresenter.java

```diff
@@ -271,41 +274,70 @@ public void onNewConnection()
    {
       // if r session bussy, fail
       if (commands_.interruptR().isEnabled()) {
-        showError(
-          "The R session is currently busy. Wait for completion or " +
-          "interrupt the current session and retry.");
-        return;
+         showError(
+            "The R session is currently busy. Wait for completion or " +
+            "interrupt the current session and retry.");
+         return;
+      }
+
+      // check for updates
+      if (!installersUpdated_) {
+         installersUpdated_ = true;
+         server_.updateOdbcInstallers(
+            new DelayedProgressRequestCallback<ConnectionUpdateResult>(
+               "Checking for Updates...") {
+
+               @Override
+               public void onSuccess(ConnectionUpdateResult result)
+               {
+                  installersWarning_ = result.getWarning();
+                  showWizard();
+               } 
+
+               @Override
+               public void onError(ServerError error)
+               {
+                  Debug.logError(error);
```

Comment:
We should show the user this error (otherwise it will just like like New Connection didn't do anything)

Reply:
Most of the server code is in a try catch, but yes, changed.